include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2021 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_HAS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_HAS_H_
+
+/**
+ * @brief Hearing Access Service (HAS)
+ *
+ * @defgroup bt_has Hearing Access Service (HAS)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * The Hearing Access Service is used to identify a hearing aid and optionally to control hearing
+ * aid presets. This API implements the server functionality.
+ *
+ * [Experimental] Users should note that the APIs can change as a part of ongoing development.
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/bluetooth.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief The preset record contains all the information about a hearing aid preset. */
+struct bt_has_preset {
+	/** @brief Preset set active callback
+	 *
+	 *  The callback can also be used locally in which case no connection will be set.
```

Comment:
How are these callbacks triggered by local actions? Side effect of `bt_has_active_preset_changed`? 

If we allow the HA server with the HAS to set these values locally, then HA need to support the CSIP Set Coordinator Role as well as the GAP Central role right?

Reply:
> How are these callbacks triggered by local actions? Side effect of `bt_has_active_preset_changed`?
I have not considered the preset name HAS (or Bluetooth in general) specific so I wanted to put it outside of HAS implementation so the value is stored somewhere else.
What local action you mean?

> If we allow the HA server with the HAS to set these values locally, then HA need to support the CSIP Set Coordinator Role as well as the GAP Central role right?
Can you elaborate?

Reply:
It is stated that:

> The callback can also be used locally in which case no connection will be set.

How are they used locally? 

My point was that if they are triggered locally, then it must mean that the value changes that trigger these callbacks, must also be possibly locally. Since HAP requires the values to be set identically for both set members, the option of setting them locally must also mean that the HA shall be able to update the other member by some mean (likely over BT), right? 

But if it is not the case that they can be changed locally by the HA, how are the callbacks used locally?

Reply:
Ah, the "locally" it's just leftover from copy-paste ^^